openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -966,15 +955,22 @@ impl UhProcessor<'_, TdxBacked> {
                     VmcsField::VMX_VMCS_EOI_EXIT_2,
                     VmcsField::VMX_VMCS_EOI_EXIT_3,
                 ];
-                for ((&field, eoi_exit), tmr) in fields
+                for ((&field, eoi_exit), (i, tmr)) in fields
                     .iter()
                     .zip(&mut self.backing.eoi_exit_bitmap)
-                    .zip(tmr.chunks_exact(2))
+                    .zip(tmr.chunks_exact(2).enumerate())
```

Comment:
Shouldn't we just zip with `self.runner.proxy_irr_exit_mut().chunks_exact(2)`?

Reply:
We borrow `runner` as mutable for the write vmcs call though, so we can't do something like this, right?
```rust
               for (((&field, eoi_exit), tmr), proxy_tmr) in fields
                    .iter()
                    .zip(&mut self.backing.eoi_exit_bitmap)
                    .zip(tmr.chunks_exact(2))
                    .zip(&mut self.runner.proxy_irr_exit_mut().chunks_exact_mut(2))
                {
                    let tmr = tmr[0] as u64 | ((tmr[1] as u64) << 32);
                    if *eoi_exit != tmr {
                        self.runner.write_vmcs64(GuestVtl::Vtl0, field, !0, tmr);
                        *eoi_exit = tmr;
                        // The kernel driver supports some common APIC functionality (ICR writes,
                        // interrupt injection). When the kernel driver handles an interrupt, it
                        // must know if that interrupt was previously level-triggered. Otherwise,
                        // the EOI will be incorrectly treated as level-triggered. We keep a copy
                        // of the tmr in the kernel so it knows when this scenario occurs.
                        proxy_tmr[0] = tmr as u32;
                        proxy_tmr[1] = (tmr >> 32) as u32;
                    }
                }
```
Yields:
```
error[E0499]: cannot borrow `*self.runner` as mutable more than once at a time
   --> openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs:966:25
    |
958 |                   for (((&field, eoi_exit), tmr), proxy_tmr) in fields
    |  _______________________________________________________________-
959 | |                     .iter()
960 | |                     .zip(&mut self.backing.eoi_exit_bitmap)
961 | |                     .zip(tmr.chunks_exact(2))
962 | |                     .zip(&mut self.runner.proxy_irr_exit_mut().chunks_exact_mut(2))
    | |_______________________________-----------_________________________________________- first borrow later used here
    |                                 |
    |                                 first mutable borrow occurs here
...
966 |                           self.runner.write_vmcs64(GuestVtl::Vtl0, field, !0, tmr);
    |                           ^^^^^^^^^^^ second mutable borrow occurs here
```
Am I missing a rust trick?

Reply:
Ah, I see. OK, then.